catalyst_voices/apps/voices/lib/widgets/text_field/voices_date_field.dart

```diff
@@ -0,0 +1,210 @@
+  void initState() {
+    final initialDate = _effectiveController.value;
+    final initialText = _convertDateToText(initialDate);
+
+    _textEditingController = TextEditingController(text: initialText);
+    _textEditingController.addListener(_handleTextChanged);
+
+    _effectiveController.addListener(_handleDateChanged);
+
+    super.initState();
+  }
+
+  @override
+  void didUpdateWidget(covariant VoicesDateField oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.controller != oldWidget.controller) {
+      (oldWidget.controller ?? _controller)?.removeListener(_handleDateChanged);
+      (widget.controller ?? _controller)?.addListener(_handleDateChanged);
+
+      final date = _effectiveController.value;
+      _textEditingController.text = _convertDateToText(date);
+    }
+  }
+
+  @override
+  void dispose() {
+    _textEditingController.dispose();
+
+    _controller?.dispose();
+    _controller = null;
+
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final onChanged = widget.onChanged;
+    final onFieldSubmitted = widget.onFieldSubmitted;
+
+    return VoicesDateTimeTextField(
+      controller: _textEditingController,
+      onChanged: onChanged != null
+          ? (value) => onChanged(_convertTextToDate(value))
+          : null,
+      validator: _validate,
+      hintText: _pattern.toUpperCase(),
+      onFieldSubmitted: onFieldSubmitted != null
+          ? (value) => onFieldSubmitted(_convertTextToDate(value))
+          : null,
+      suffixIcon: ExcludeFocus(
+        child: VoicesIconButton(
+          onTap: widget.onCalendarTap,
+          child: VoicesAssets.icons.calendar.buildIcon(),
+        ),
+      ),
+      borderRadius: widget.borderRadius,
+      dimBorder: widget.dimBorder,
+    );
+  }
+
+  void _handleTextChanged() {
+    final date = _convertTextToDate(_textEditingController.text);
+    if (_effectiveController.value != date) {
+      _effectiveController.value = date;
+    }
+  }
+
+  void _handleDateChanged() {
+    final text = _convertDateToText(_effectiveController.value);
+    if (_textEditingController.text != text) {
+      _textEditingController.text = text;
+    }
+  }
+
+  String _convertDateToText(DateTime? value) {
+    if (value == null) {
+      return '';
+    }
+
+    final day = value.day.toString().padLeft(2, '0');
+    final month = value.month.toString().padLeft(2, '0');
+    final year = value.year;
+
+    return '$day/$month/$year';
+  }
+
+  DateTime? _convertTextToDate(String value) {
+    if (value.isEmpty) return null;
+
+    final parts = value.split('/');
+    if (parts.length != 3) return null;
+
+    try {
+      final reformatted = '${parts[2]}-${parts[1]}-${parts[0]}';
+      final formatDt = DateTime.parse(reformatted);
+
+      if (formatDt.month < 1 || formatDt.month > 12) return null;
+      if (formatDt.day < 1 || formatDt.day > 31) return null;
+      if (formatDt.year < 1900 || formatDt.year > 2100) return null;
```

Comment:
DateTime.parse using overflow method so this if'es won't work. Eg. the user puts 32/12/2024 DateTime.parse make it 01/01/2025